apps/spruce/src/pages/configurePatch/configurePatchCore/useConfigurePatch/index.ts

```diff
@@ -1,169 +1,58 @@
 import { useEffect, useReducer } from "react";
```

Comment:
Large refactor on this hook. Primarily extracted state logic to a different file and reworked how tabs are referenced so that we reference the tab name instead of an index